File: src/test/java/com/github/xumeng/java/problems/easy/problem0345/Solution0345Test.java
package com.github.xumeng.java.problems.easy.problem0345;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0345Test {

    private final Solution0345 solution = new Solution0345();

    @Test
    public void test1() {
        String s = "IceCreAm";
        String result = solution.reverseVowels(s);
        String expected = "AceCreIm";
        assertEquals(expected, result);
    }

    @Test
    public void test2() {
        String s = "leetcode";
        String result = solution.reverseVowels(s);
        String expected = "leotcede";
        assertEquals(expected, result);
    }

    @Test
    public void test3() {
        String s = "a";
        String result = solution.reverseVowels(s);
        String expected = "a";
        assertEquals(expected, result);
    }

    @Test
    public void test4() {
        String s = "b";
        String result = solution.reverseVowels(s);
        String expected = "b";
        assertEquals(expected, result);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0345/Solution0345Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0345

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0345Test {

    private val solution = Solution0345()

    @Test
    fun test1() {
        val s = "IceCreAm"
        val result = solution.reverseVowels(s)
        val expected = "AceCreIm"
        assertEquals(expected, result)
    }

    @Test
    fun test2() {
        val s = "leetcode"
        val result = solution.reverseVowels(s)
        val expected = "leotcede"
        assertEquals(expected, result)
    }

    @Test
    fun test3() {
        val s = "a"
        val result = solution.reverseVowels(s)
        val expected = "a"
        assertEquals(expected, result)
    }

    @Test
    fun test4() {
        val s = "b"
        val result = solution.reverseVowels(s)
        val expected = "b"
        assertEquals(expected, result)
    }

}